packages/repository/src/mixins/repository.mixin.ts

```diff
@@ -255,6 +286,24 @@ export function RepositoryMixin<T extends MixinTarget<Application>>(
   };
 }
 
+/**
+ * This interface describes additional Component properties
+ * allowing components to contribute Repository-related artifacts.
+ */
+export interface RepositoryComponent {
+  /**
+   * An optional list of Repository classes to bind for dependency injection
+   * via `app.repository()` API.
+   */
+  repositories?: Class<Repository<Model>>[];
+
+  /**
+   * An optional list of Model classes to bind for dependency injection
+   * via `app.model()` API.
+   */
+  models?: Class<Model>[];
```

Comment:
Do we want to add `dataSources`?

Reply:
I am not sure. DataSources typically require environment-specific configuration (e.g. database connection string), I am no sure if it makes sense to export the from a component. Either way, such change is out of scope of this pull request - feel free to open a follow-up GH issue to discuss this idea more.